src/test/java/com/michalsydoryk/app/board/Board2DTest.java

```diff
@@ -0,0 +1,159 @@
+package com.michalsydoryk.app.board;
+
+import com.michalsydoryk.app.coordinates.Coordinates2D;
+import com.michalsydoryk.app.sign.Sign;
+import org.testng.Assert;
+import org.testng.annotations.*;
+
+import java.util.ArrayDeque;
+import java.util.HashSet;
+import java.util.Queue;
+import java.util.Set;
+
+@Test
+public class Board2DTest {
```

Comment:
Generally most of tests can be parametrized.

Reply:
will be fixed